src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,200 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValid(index, true);
+        Node<T> oldNode = findNodeByIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0 && size != 0) {
+            addBeforeHead(newNode);
+        } else if (index == 0) {
+            add(value);
+        } else if (index == size) {
+            addAfterTail(newNode);
+        } else if (oldNode != null) {
```

Comment:
is oldNode could be null is case is index valid? What if oldNode is null?

Reply:
My bad, I have just understood that it seemed to be weird in the case